File: backend/src/auth/auth.controller.ts
import {
  Controller,
  Post,
  Body,
  HttpException,
  HttpStatus,
} from '@nestjs/common';
import { AuthService } from './auth.service';
import { LoginDto, ChangePasswordDto, AuthResponse } from './dto/auth.dto';

@Controller('auth')
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @Post('login')
  login(@Body() loginDto: LoginDto): AuthResponse {
    try {
      const result = this.authService.validateAdmin(
        loginDto.username,
        loginDto.password,
      );

      if (!result.success) {
        throw new HttpException(
          result.message || '登录失败',
          HttpStatus.UNAUTHORIZED,
        );
      }

      return { success: true, message: '登录成功' };
    } catch (error) {
      if (error instanceof HttpException) {
        throw error;
      }

      throw new HttpException(
        '服务器内部错误',
        HttpStatus.INTERNAL_SERVER_ERROR,
      );
    }
  }

  @Post('change-password')
  changePassword(@Body() changePasswordDto: ChangePasswordDto): AuthResponse {
    try {
      const result = this.authService.changePassword(
        changePasswordDto.currentPassword,
        changePasswordDto.newPassword,
      );

      if (!result.success) {
        throw new HttpException(
          result.message || '密码修改失败',
          HttpStatus.BAD_REQUEST,
        );
      }

      return { success: true, message: result.message };
    } catch (error) {
      if (error instanceof HttpException) {
        throw error;
      }

      throw new HttpException(
        '服务器内部错误',
        HttpStatus.INTERNAL_SERVER_ERROR,
      );
    }
  }
}

File: backend-dotnet/BlogApi/Controllers/AuthController.cs
using Microsoft.AspNetCore.Mvc;
using BlogApi.Services;
using BlogApi.DTOs;

namespace BlogApi.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class AuthController : ControllerBase
    {
        private readonly AuthService _authService;

        public AuthController(AuthService authService)
        {
            _authService = authService;
        }

        [HttpPost("login")]
        public ActionResult<AuthResponse> Login([FromBody] LoginDto dto)
        {
            try
            {
                var result = _authService.ValidateAdmin(dto.Username, dto.Password);

                if (!result.Success)
                {
                    return Unauthorized(result);
                }

                return Ok(result);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new AuthResponse
                {
                    Success = false,
                    Message = $"服务器内部错误: {ex.Message}"
                });
            }
        }

        [HttpPost("change-password")]
        public ActionResult<AuthResponse> ChangePassword([FromBody] ChangePasswordDto dto)
        {
            try
            {
                var result = _authService.ChangePassword(dto.CurrentPassword, dto.NewPassword);

                if (!result.Success)
                {
                    return BadRequest(result);
                }

                return Ok(result);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new AuthResponse
                {
                    Success = false,
                    Message = $"服务器内部错误: {ex.Message}"
                });
            }
        }
    }
}
